### PR TITLE
Policy related Changes are moved to CXF 3.4

### LINKED ISSUE #20439: xmlWS-3.0: Skip policy and skip attachment settings
These 2 functionalities are already set at CXF 2.6.2 bundles, this issue is to move these changes to CXF 3.4 version.

### PR DESCRIPTION
**Pull request by BerksanAtes**
Fixes #20439
Addition of new source code in IBM repositories triggered FFDC creation on com.ibm.ws.wssecurity_fat.wsscxf.1 FAT bucket. This issue is resolved by allowing these FFDCs.

### CODE REVIEW

**Comment by BerksanAtes** [comment]
#build

**Comment by BerksanAtes** [comment]
#build

**Comment by BerksanAtes** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.wssecurity_fat.wsscxf,com.ibm.ws.wssecurity_fat.wsscxf.1

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxws.3.2/src/org/apache/cxf/jaxws/interceptors/SwAOutInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+
+public class SwAOutInterceptor extends AbstractSoapInterceptor {
+    private static final Logger LOG = LogUtils.getL7dLogger(SwAOutInterceptor.class);
+
+    private static final Map<String, Method> SWA_REF_METHOD
+        = new ConcurrentHashMap<>(4, 0.75f, 2);
+    private static final Set<String> SWA_REF_NO_METHOD
+        = Collections.newSetFromMap(new ConcurrentHashMap<String, Boolean>(4, 0.75f, 2));
+
+//  Begin copied over from CXF 2.6.2
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxws.3.2/src/org/apache/cxf/jaxws/interceptors/SwAOutInterceptor.java]: Can you mark with `LIberty Change` as well? Should follow the format of the others.

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxws.3.2/src/org/apache/cxf/jaxws/interceptors/SwAOutInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+
+        Collection<Attachment> atts = message.getAttachments();
+        if (atts == null) {
+            atts = new ArrayList<>();
+            message.setAttachments(atts);
+        }
+        return atts;
+    }
+    
+//  Begin copied over from CXF 2.6.2
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxws.3.2/src/org/apache/cxf/jaxws/interceptors/SwAOutInterceptor.java]: Same as above, include `Liberty Change`

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyEngineImpl.java [hunk 3]
```diff
@@ -0,0 +1,702 @@
...
+    private boolean enabled = true;
+    private Boolean ignoreUnknownAssertions;
+    private boolean addedBusInterceptors;
+    private AlternativeSelector alternativeSelector;
+    
+    private static boolean ignoreUnsupportedPolicy;
+    
+    static {
+
+        String skipPolicyCheck = System.getProperty("cxf.ignore.unsupported.policy");
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyEngineImpl.java:88]: Add `Liberty Change`  to this

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyVerificationInInterceptor.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+ */
+public class PolicyVerificationInInterceptor extends AbstractPolicyInterceptor {
+    public static final PolicyVerificationInInterceptor INSTANCE = new PolicyVerificationInInterceptor();
+
+    private static final Logger LOG = LogUtils.getL7dLogger(PolicyVerificationInInterceptor.class);
+
+    private static boolean ignoreUnsupportedPolicy;
+    
+    static {
+
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyVerificationInInterceptor.java:52]: Add `Liberty Change` to this

File: dev/com.ibm.ws.wssecurity_fat.wsscxf.1/fat/src/com/ibm/ws/wssecurity/fat/cxf/x509migtoken/CxfX509MigTests.java [hunk 5]
```diff
@@ -2939,6 +2939,7 @@ public void testBadCxfX509AsymIssuerSerialMigService() throws Exception {
      */
 
     @Test
+    @AllowedFFDC(value = { "org.apache.cxf.ws.policy.PolicyException" }, repeatAction = { JakartaEE9Action.ID })
```
Comment (neuwerk) [inline dev/com.ibm.ws.wssecurity_fat.wsscxf.1/fat/src/com/ibm/ws/wssecurity/fat/cxf/x509migtoken/CxfX509MigTests.java]: I'm concerned about these changes. It doesn't seem correct that xmlWS-3.0's tests generate one FFDC, when the other runs generate a different FFDC. Updating the the FFDC can be OK in certain situations, but I don't think this `@AllowedFFDC` should be necessary. I think we need to investigate why one is getting one version and the other is getting another.

**Review by neuwerk — CHANGES_REQUESTED** [review]

**Review by sawadood — APPROVED** [review]
The property changes look good!

**Review by neuwerk — APPROVED** [review]

**Comment by BerksanAtes** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.wssecurity_fat.wsscxf,com.ibm.ws.wssecurity_fat.wsscxf.1

**Comment by BerksanAtes** [comment]
#run-libby-bot

**Pull request merged**